libs/components/indicators/src/lib/modules/alert/alert.component.ts

```diff
@@ -10,69 +16,60 @@ const ALERT_TYPE_DEFAULT = 'warning';
   selector: 'sky-alert',
   styleUrls: ['./alert.component.scss'],
   templateUrl: './alert.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SkyAlertComponent implements OnInit {
+export class SkyAlertComponent {
   // TODO: Change alertType to SkyIndicatorIconType in a breaking change.
   /**
    * Specifies a style for the alert to determine the icon and background color.
    * The valid options are `danger`, `info`, `success`, and `warning`.
    * @default "warning"
    */
   @Input()
-  public set alertType(value: string) {
-    this._alertType = value;
-    this.alertTypeOrDefault =
-      (value as SkyIndicatorIconType) || ALERT_TYPE_DEFAULT;
-    this.updateAlertIcon();
+  public set alertType(value: string | undefined) {
+    this.#_alertType = value;
+    if (this.alertType !== this.alertTypeOrDefault) {
+      this.alertTypeOrDefault =
+        (this.alertType as SkyIndicatorIconType) || ALERT_TYPE_DEFAULT;
+      this.indicatorIcon = SkyIndicatorIconUtility.getIconsForType(
+        this.alertTypeOrDefault
+      );
```

Comment:
We update the icon values every time there is an update to `alertTypeOrDefault`.